pkgs/top-level/options.nix

```diff
@@ -0,0 +1,128 @@
+{lib, stdenv}:
```

Comment:
Can you use `hostPlatform` instead?

Reply:
Yeah definitely! targetPlatform might actually make more sense though. In fact, one of the big reasons for this is getting rid of a lot of the ```stdenv.isDarwin``` and ```stdenv.isLinux``` that is spread accross Nixpkgs. Quite a few should be easily replaced with ```withApple```, ```withSystemd``` and ```withAlsa```.